### PR TITLE
MOBT-353: Spot-extract uses resample percentiles to produce unavailable percentiles

### PR DESCRIPTION
**Pull request by bayliffe**
Modifies the spot-extract CLI to use ResamplePercentiles when necessary. This is required when spot-extracting percentile output from gridded percentile inputs that do not contain the target percentiles.

Previously the CLI extracted as much as it could and only raised an exception if no data was available. Following this PR the CLI will instead check if the desired percentiles are available within the gridded data and perform a simple extraction if they are. If not it will resample the percentiles to generate the target set. No exception is now raised as the target percentiles can always be produced (with no doubt variable quality if the input percentile set is insufficient).

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks @bayliffe 👍 

I've added one docstring comment and a query.

File: improver/cli/spot_extract.py [hunk 1]
```diff
@@ -156,7 +157,7 @@ def process(
     # to minimise processing time; usually there are far fewer spot sites than
     # grid points.
     if extract_percentiles:
-        extract_percentiles = [np.float32(x) for x in extract_percentiles]
+        extract_percentiles = sorted([np.float32(x) for x in extract_percentiles])
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/cli/spot_extract.py:160]: It's legacy now and could be removed. When I modified the code to raise an exception if less than all of the requested percentiles were returned I did so using the length of the returned percentile coordinate. You suggests comparing the values themselves, which I did. However, whilst the user can specify percentiles in any order they like, the returned cube coordinate will be monotonically increasing. Therefore to compare the arrays simply I sorted the requested percentiles as well; see [this commit](https://github.com/metoppv/improver/pull/1796/commits/5c5ba8f23ad05126eb7635e44a779d6acf0fb3d9).

As we are no longer performing this check the test and sorting have no real value. Still, I might leave the disordered test just to demonstrate that the user specified order is irrelevant. I'll remove the `sorted` statement from the code as it is not needed.

File: improver/cli/spot_extract.py [hunk 2]
```diff
@@ -130,6 +130,7 @@ def process(
 
     from improver.ensemble_copula_coupling.ensemble_copula_coupling import (
         ConvertProbabilitiesToPercentiles,
+        ResamplePercentiles,
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/cli/spot_extract.py:133]: Done.

**Review by gavinevans — DISMISSED** [review]

**Review by mspelman07 — CHANGES_REQUESTED** [review]
One small comment about a doc string.

File: improver/cli/spot_extract.py [hunk 3]
```diff
@@ -130,6 +131,7 @@ def process(
...
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/cli/spot_extract.py:129]: Good spot. Removed.

**Review by mspelman07 — APPROVED** [review]

**Pull request merged**